kiwi/bootloader/config/base.py

```diff
@@ -185,6 +186,17 @@ def failsafe_boot_entry_requested(self):
             return False
         return True
 
+    def mediacheck_boot_entry_requested(self):
+        """
+        Check if a mediacheck boot entry is requested
+
+        :rtype: bool
+        """
+        if self.xml_state.build_type.get_checkiso() is True and \
+                platform.machine() in ['x86_64', 'i586', 'i686']:
+            return True
+        return False
+
```

Comment:
Yes you are right, I haven't done it this way because I wanted to have a slightly different check in builder/live.py. But if we include the runtime check to just fail in non x86 architectures then this is fine, the same xml_state method to get the isocheck could be used in both, builder/live.py and in this bootloader interface.